# import statements

In [5]:
import numpy as np
import pandas as pd

# code to read student.csv into pandas dataframe 

In [11]:
file = pd.read_csv('student.csv', delimiter=';')

# helper functions (if needed)

In [13]:
print(file)

    school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
1       GP   F   17       U     GT3       T     1     1   at_home     other   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
390     MS   M   20       U     LE3       A     2     2  services  services   
391     MS   M   17       U     LE3       T     3     1  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
393     MS   M   18       R     LE3       T     3     2  services     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     ... famrel freetime  goout  Dalc  Walc health 

# Python code to answer question 1) What is the minimum, maximum, average, and median final grade (G3 column)?

In [14]:
min_grade = file['G3'].min()
max_grade = file['G3'].max()
avg_grade = file['G3'].mean()
median_grade = file['G3'].median()

print("Min Grade: ", min_grade)
print("Max Grade: ", max_grade)
print("Average Grade: ", round(avg_grade, 2))
print("Median Grade: ", median_grade)

Min Grade:  0
Max Grade:  20
Average Grade:  10.42
Median Grade:  11.0


# Text answer for question 1.

Minimum Grade:  0
Maximum Grade:  20
Average Grade:  10.42
Median Grade:  11.0

# Python code to answer question 2) Which subgroup of sex, male or female, had the highest average final grade (G3)?

In [15]:
sex_group = file.groupby('sex')
avg_grade_b_sex = sex_group['G3'].mean()
highest_avg_sex = avg_grade_b_sex.idxmax()

print("Subgroup with highest average grade is: ", highest_avg_sex)

Subgroup with highest average grade is:  M


# Text answer for question 2.  

SubGroup with the highest average final grade is: Male

# Python code to answer question 3) Which subgroup of Fjob had the lowest average final grade?

In [16]:
fjob_group = file.groupby('Fjob')
avg_grade_fjob = fjob_group['G3'].mean()
low_avg_fjob = avg_grade_fjob.idxmin()

print("Subgroup with lowest average grade: ", low_avg_fjob)

Subgroup with lowest average grade:  at_home


# Text answer for question 3.

Subgroup with the lowest average final grade is: At Home

# Python code to answer question 4) On average, did students in romantic relationships get higher or lower final grades?

In [20]:
#python code...

# Text answer for question 4.

text...

# Python code to answer question 5) On average, do people with internet access at home make higher or lower final grades than people without internet? 

In [21]:
#python code...

# Text answer for question 5.

text...

# Python code to answer question 6) What percentage of male students are in romantic relationships?  What percentage of female students are in romantic relationships?

In [22]:
#python code...

# Text answer for question 6.

text...

# Python code to answer question 7) What is the correlation between freetime and final grade? 

In [23]:
#python code...

# Text answer for question 7.

text...

# Python code to answer question 8) What is the correlation between studytime and final grade? 

In [24]:
#python code...

# Text answer for question 8.

text...

# Python code to answer question 9) What is the correlation between studytime and freetime? 

In [25]:
#python code...

# Text answer for question 9.

text...

# Python code to answer question 10) Excluding columns G1 and G2, which numerical column had the biggest correlation with final grade (G3)?

In [26]:
#python code...

# Text answer for question 10.  Please identify the column and correlation.

text...